src/provider/mod.rs

```diff
@@ -109,17 +109,35 @@ pub(crate) enum BlobOrCollection {
 }
 
 impl BlobOrCollection {
-    pub fn is_blob(&self) -> bool {
+    pub(crate) fn is_blob(&self) -> bool {
         matches!(self, BlobOrCollection::Blob { .. })
     }
 
-    pub fn blob_path(&self) -> Option<&Path> {
+    pub(crate) fn blob_path(&self) -> Option<&Path> {
         match self {
             BlobOrCollection::Blob { path, .. } => Some(path),
             BlobOrCollection::Collection { .. } => None,
         }
     }
 
+    #[allow(dead_code)]
```

Comment:
why?

Reply:
at some point it was dead code I guess, or it was only used in test. Not sure anymore. Will remove.